report_gen/README.md

```diff
@@ -0,0 +1,157 @@
+
+Ignores the first line of the csv file, as it commonly holds the name of the command which generated it.
+
+Output filenames are of the format: all:x_axis-y_axis.extension
+
+####CSV files used in example:
+
+    a.csv:                 b.csv:                 c.csv:
+    
+    firstline              firstline              firstline
+    x0, y1, y2, y3         y1, x0, y2, z4         x0, y1, y3, z4
+    1,  2,  8,  4          1,  1,  9,  5          1,  4,  8,  6
+    2,  4,  4,  3          2,  3,  5,  4          2,  6,  4,  5
+    3,  8,  2,  2          4,  7,  3,  3          3,  7,  2,  4
+
+As seen from the structure of the csv files:  
+1. The order of columns can be different from file to file  
+2. Not all files have to contain the same columns
+
+#### Example usage:
+    python graphgen.py -f example/a.csv example/b.csv example/c.csv -o example/figs/graphgen
+
+#### Generated graphs:  
+
+|example/figs/graphgen/all:x0-y1.svg|example/figs/graphgen/all:x0-y2.svg
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y1.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-y2.svg" width="300" height="auto">
+
+|example/figs/graphgen/all:x0-y1.svg|example/figs/graphgen/all:x0-y2.svg
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y3.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-z4.svg" width="300" height="auto">
+
+### Features
+#### 1. Plot by category
+
+Plots graphs by category: for each test separately, parses columns for search strings and plots them with a common name on the y axis. The search string and the name are input arguments in dict form: '{"searchstring":"name"}'. 
+
+Output filenames are of the format: benchmark_name:x_axis-category_name.extension 
+
+##### Example:
+	python graphgen.py -f example/a.csv example/b.csv example/c.csv -o example/figs/graphgen -c '{"y":"Ytitle","z":"Ztitle"}'
+
+##### Generated graphs: 
+ 
+Graphs generated by the basic functionality, and:
+
+|example/figs/graphgen/a:x0-Ytitle.svg|example/figs/graphgen/b:x0-Ytitle.svg|example/figs/graphgen/c:x0-Ytitle.svg|
+:-------------------------:|:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/a:x0-Ytitle.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/b:x0-Ytitle.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/c:x0-Ytitle.svg" width="300" height="auto">
+
+|example/figs/graphgen/b:x0-Ztitle.svg|example/figs/graphgen/c:x0-Ztitle.svg|
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/b:x0-Ztitle.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/c:x0-Ztitle.svg" width="300" height="auto">
+
+#### 2. Plot on multiple custom x axes
+
+The input x axes do not have to match with the columns perfectly; e.g. -x y would plot with x axes y1,y2,y3.
```

Comment:
I did not get what it does from this sentence. I would call that "custom x axes" and say that with this argument you can select source data for x axes...